File: server/internal/trip/trip.go
package trip

import (
	"github.com/huydnt1801/chuyende/internal/ent/trip"
)

type Trip struct {
	ID            int         `json:"id"`
	UserID        int         `json:"userId"`
	DriveID       int         `json:"driveId,omitempty" mapstructure:"driveId,omitempty"`
	StartX        float64     `json:"startX"`
	StartY        float64     `json:"startY"`
	StartLocation string      `json:"startLocation"`
	EndX          float64     `json:"endX"`
	EndY          float64     `json:"endY"`
	EndLocation   string      `json:"endLocation"`
	Distance      float64     `json:"distance"`
	Type          trip.Type   `json:"type"`
	Price         float64     `json:"price"`
	Status        trip.Status `json:"status,omitempty"`
	Rate          int         `json:"rate,omitempty" mapstructure:"rate,omitempty"`
}

type TripParams struct {
	TripID  *int
	UserID  *int
	DriveID *int
	Status  *trip.Status
	Rate    *int
}

type TripUpdate struct {
	DriveID *int
	Status  *trip.Status
	Rate    *int
}

File: mobile/src/api/Trip.js
import axios from "axios";
import Utils from "../share/Utils";
import baseApi from "./baseApi";

const getList = () => {
    const url = "/trips";
    return baseApi.get(url, {
        headers: {
            cookie: Utils.global.cookie
        }
    });
}

/**
 * @typedef params
 * @property {Number} startX 
 * @property {Number} startY 
 * @property {String} startLocation 
 * @property {Number} endX 
 * @property {Number} endY 
 * @property {String} endLocation 
 * @property {Number} distance 
 * @property {Number} distance 
 * @property {String} type 
 * @param {params} params
 * @returns 
 */
const createTrip = (params) => {
    const url = "/trips";
    const param = {
        ...params
    }
    return baseApi.post(url, param, {
        headers: {
            cookie: Utils.global.cookie
        }
    });
}


const checkPrice = (distance) => {

    const url = "/trips/price";
    const params = {
        distance: distance
    }

    return baseApi.get(url, {
        params,
        headers: {
            cookie: Utils.global.cookie
        }
    });
}

const rate = (tripId, star) => {
    const url = `trips/${tripId}/rate`;
    const param = {
        rate: star
    }

    return baseApi.patch(url, param, {
        headers: {
            cookie: Utils.global.cookie
        }
    });
}

const updateStatus = (tripId, status) => {
    const url = `trips/${tripId}/status`;
    const param = {
        status: status
    }

    return baseApi.patch(url, param, {
        headers: {
            cookie: Utils.global.cookie
        }
    });
}

const tripApi = {
    getList,
    createTrip,
    checkPrice,
    rate,
    updateStatus
}

export default tripApi